evo1/atb/entity.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from enum import Enum, auto
+
+from evo1.memory.atb import BattleEntity
+
+logger = logging.getLogger(__name__)
+
+
+class ATBEntityID(Enum):
+    CLINK = auto()
+    KAERIS = auto()
+    # Bosses
+    KEFKAS_GHOST = auto()
+    ZEPHYROS = auto()
+    # Overworld 2D
+    SLIME = auto()
+    EMUK = auto()
+    # Cavern
+    SCAVEN = auto()
+    KOBRA = auto()
+    TORK = auto()
+    # Overworld 3D
+    ZOOMBA = auto()
+    APIDYA = auto()
+    ATUIN = auto()
+
+
+class ATBEntity:
+    def __init__(self, kind: ATBEntityID, turn_gauge: float) -> None:
+        self.kind = kind
+        self.turn_gauge = turn_gauge
+
+
+class ATBEntityStats:
+    def __init__(
+        self,
+        max_hp: int,
+        cur_hp: int,
+        attack: int,
+        defense: int,
+        magic: int,
+        evade: int,
+    ) -> None:
+        self.cur_hp = cur_hp
+        self.max_hp = max_hp
+        self.attack = attack
+        self.defense = defense
+        self.magic = magic
+        self.evade = evade
+
+
+def atb_stats_from_id(entity_id: ATBEntityID, level: int = 0) -> ATBEntityStats:
```

Comment:
The reason for this function is that out of combat, we don't have access to the stat blocks of the party or enemies. They are probably in memory somewhere, but I've so far not been able to find them. So some hardcoded stuff here.

Reply:
I feel like objects would be a better solution for the guts of this function, but that's probably because I don't like passing around multiple variables.

But my idea would also require pulling most of these stats out of memory rather than hard-coding them, and since we may or may not have easy access to that, not really worth trying to tinker with anyhting.